web-client/src/presenter/computeds/Public/publicTrialSessionsHelper.ts

```diff
@@ -81,6 +88,27 @@ function groupTrialSessions(
 
 const PAGE_SIZE = 100;
 
+function dateComparison(tsGroup): boolean {
+  const firstEndDate = tsGroup?.rows?.[0]?.formattedEstimatedEndDate;
+
+  const todayIso = createISODateAtStartOfDayEST();
+  let compareIso;
+
+  if (firstEndDate) {
+    const firstEndIso = createISODateString(firstEndDate, FORMATS.MMDDYYYY);
+
+    compareIso = createISODateAtStartOfDayEST(firstEndIso);
+  } else {
+    const firstStartDate = tsGroup?.rows?.[0]?.startDate;
+
+    if (!firstStartDate) return false;
+
+    compareIso = createISODateAtStartOfDayEST(firstStartDate);
+  }
+
+  return dateStringsCompared(compareIso, todayIso) >= 0;
+}
+
```

Comment:
```suggestion
function dateComparison(tsGroup): boolean {
  const firstRow = tsGroup?.rows?.[0];
  if (!firstRow) return false;

  const dateToCompare = firstRow.formattedEstimatedEndDate || firstRow.startDate;
  if (!dateToCompare) return false;

  const todayIso = createISODateAtStartOfDayEST();
  
  const compareIso = firstRow.formattedEstimatedEndDate
    ? createISODateAtStartOfDayEST(createISODateString(dateToCompare, FORMATS.MMDDYYYY))
    : createISODateAtStartOfDayEST(dateToCompare);

  return dateStringsCompared(compareIso, todayIso) >= 0;
}
```
I ran this through ai to optimize it. It's up to you if you want to take it or not, but I think its easier to understand.